### PR TITLE
Updated the Beginners Section of Terraform

### PR DESCRIPTION
**Pull request by knoxknot**
Answered the definition questions for provider, resources and provisioner and those of terraform commands.

### CODE REVIEW

**Comment by bregman-arie** [comment]
You added the answers in the questions. Answers should be added after "<br><b>"
Thanks for contributing.

**Comment by bregman-arie** [comment]
Sorry, still not valid. See an example of questions and answer:

<details>
<summary>Explain stateless vs. stateful</summary><br><b>
 
Stateless applications don't store any data in the host which makes it ideal for horizontal scaling and microservices.                                                       
Stateful applications depend on the storage to save state and data, typically databases are stateful applications.
</b></details>


I also suggest using some editor where you can preview the result before uploading the change. Thank again.

File: README.md [hunk 1]
```diff
@@ -1609,11 +1609,14 @@ The benefits of Terraform over the other tools:
 <details>
 <summary>Explain each of the following:
```
Comment (surister) [inline README.md]: Summary starts here, summary tag defines the question.

File: README.md [hunk 2]
```diff
@@ -1609,11 +1609,14 @@ The benefits of Terraform over the other tools:
...
 
-  * Provider
-  * Resource
-  * Provisioner
-</b></details>
-
+  * Provider: 
+  * Resource: 
+  * Provisioner: 
+</b></details><br><b>
```
Comment (surister) [inline README.md]: Here summary should be closed in order to wrapp the question.

Comment (surister) [reply README.md]: Also details should not be closed here. As @bregman-arie  showed you, details wraps the entire question + answer whereas summary only wraps the question. You can check your own fork to check whether the changes are alright or not.
https://github.com/knoxknot/devops-interview-questions/blob/master/README.md#terraform-beginner

**Comment by knoxknot** [comment]
The mentioned edits are effected. Please review again. Thanks

**Review by surister — APPROVED** [review]
Seems good

File: README.md [hunk 3]
```diff
@@ -1625,12 +1628,15 @@ It keeps track of the IDs of created resources so that Terraform knows what it i
...
-  * <code>terraform validate</code> parses through the terraform code to ensure that they are both syntactically correct and runtime 
-  				    error free.
-  * <code>terraform apply</code> this command provision's the described services on the providers platform.
+  * <code>terraform init</code> 
+  * <code>terraform plan</code>	
+  * <code>terraform validate</code> 
+  * <code>terraform apply</code> 
+  <details> - terraform init initialises and downloads the provider's binary as described in the .tf files
+ - terraform plan is a dry-run listing the services and components that will be provisioned on specified provider's platform.
+- terraform validate parses through the terraform code to ensure that they are both syntactically correct and runtime error free.
```
Comment (surister) [inline README.md]: Although, seems that `terraform validate` that exist in this commit was lost in recent change :P

**Comment by bregman-arie** [comment]
I believe we can merge it now.
Two tips for next PRs (to any project):

* Squash the commits into one or two commits. Right now I'm going to merge the 8 commits you submitted although it's a small change because I don't want to discourage anyone from learning and contributing
* I don't know what editor you are using (I use simple vi) but for changes that affect the "UI" part of the project, you might want to use editor with preview mode so you can see how your code affects the project from an appearance perspective

Thank you again for contributing!

**Pull request merged**